modules/light-clients/06-solomachine/update.go

```diff
@@ -44,7 +44,25 @@ func (cs ClientState) verifyHeader(ctx sdk.Context, cdc codec.BinaryCodec, clien
 	}
 
 	// assert currently registered public key signed over the new public key with correct sequence
-	data, err := HeaderSignBytes(cdc, header)
+	headerData := &HeaderData{
+		NewPubKey:      header.NewPublicKey,
+		NewDiversifier: header.NewDiversifier,
+	}
+
+	dataBz, err := cdc.Marshal(headerData)
+	if err != nil {
+		return err
+	}
+
+	signBytes := &SignBytesV2{
+		Sequence:    header.Sequence,
```

Comment:
Given:
```go
// assert update sequence is current sequence
	if header.Sequence != cs.Sequence {
		return sdkerrors.Wrapf(
			clienttypes.ErrInvalidHeader,
			"header sequence does not match the client state sequence (%d != %d)", header.Sequence, cs.Sequence,
		)
}
```

I wonder if we should just remove the Sequence argument from the header proto type and just use the current client state sequence

Reply:
I'd be happy to do that in this series of PRs if you think its safe to omit in the `Header` proto.